### PR TITLE
feat: add DialUpgrade to upgrade passed tcp connection to ssl

### PR DESCRIPTION
**Pull request by mikaelthd**
Needed for ME-5

Signed-off-by: Mikael Torp-Holte <mikael.torp-holte@northern.tech>

### CODE REVIEW

File: net.go [hunk 1]
```diff
@@ -149,3 +149,54 @@ func DialSession(network, addr string, ctx *Ctx, flags DialFlags,
 	}
 	return conn, nil
 }
+
+// DialUpgrade will make the connection conn the underlying connection of an
+// OpenSSL client connection using context ctx.
```
Comment (lluiscampos) [inline net.go]: This sentence is very difficult to read

File: net.go [hunk 2]
```diff
@@ -149,3 +149,54 @@ func DialSession(network, addr string, ctx *Ctx, flags DialFlags,
...
+// If flags includes InsecureSkipHostVerification, the server certificate's
+// hostname will not be checked to match the hostname in addr. Otherwise, flags
+// should be 0.
+//
+// Dial probably won't work for you unless you set a verify location or add
```
Comment (lluiscampos) [inline net.go]:
```suggestion
// DialUpgrade probably won't work for you unless you set a verify location or add
```
?

File: net.go [hunk 3]
```diff
@@ -149,3 +149,54 @@ func DialSession(network, addr string, ctx *Ctx, flags DialFlags,
...
+// some certs to the certificate store of the client context you're using.
+// This library is not nice enough to use the system certificate store by
+// default for you yet.
+func DialUpgrade(conn net.Conn, addr string, ctx *Ctx,
```
Comment (lluiscampos) [inline net.go]: Nitpick: I would name this one
```suggestion
func DialUpgrade(tcpConn net.Conn, addr string, ctx *Ctx,
```

and the created SSL connection just `conn`.

This is to be consistent with previous method `DialSession`

**Review by lluiscampos — DISMISSED** [review]

File: net.go (hunk 2 above)
Comment (mikaelthd) [reply net.go]: Haha, yes but then
https://github.com/mendersoftware/openssl/blob/9fe59ddd6ba46e4030a300bfc9363de62e437786/net.go#L93-L100

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**